### PR TITLE
fixes #8435 - progress bar on repository content-upload

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

**Comment by ehelms** [comment]
Does this introduce powerbar as a new dependency? Or does hammer-cli already provide this? If not, will we need to package this or is it already packaged?

**Comment by thomasmckay** [comment]
@ehelms powerbar is already present and used for async task output.

**Comment by ehelms** [comment]
Oh, from hammer-cli-foreman-tasks ? You'll want to require this in the gemspec just it does. 

There is a question of whether it would be worth creating a module in hammer-cli that encompasses this behavior so that we only have to override what is specific to a given task that wants a progress bar.

**Comment by thomasmckay** [comment]
Added powerbar to the spec. I considered breaking it out into a utility but opted not to since the amount of shared code was small. I figured we could revisit if the pattern came up again.

File: lib/hammer_cli_katello/repository.rb [hunk 1]
```diff
@@ -255,6 +262,27 @@ def import_upload_ids(ids)
         }
         resource.call(:import_uploads, params)
       end
+
+      def progress_message(filename)
+        _('Uploading %{filename}') % {:filename => filename}
+      end
+
+      def render_result
+        print_message _('finished')
```
Comment (ehelms) [inline lib/hammer_cli_katello/repository.rb]: Looks like the trend is capitalized words and what is the effect if errors occur during the upload process? Since the progress bar method is doing an 'ensure' this finished would get printed everytime, but will it be obvious there were errors and what they are?

Comment (thomasmckay) [inline lib/hammer_cli_katello/repository.rb]: Not sure I want to expand this pull-request into more than what's here. Perhaps this was not the "low hanging fruit" issue I had hoped it was. :)

Comment (ehelms) [inline lib/hammer_cli_katello/repository.rb]: I don't think what I am asking is all that much, just to check/ensure that if there is an error state the user is presented with that information.

Comment (thomasmckay) [inline lib/hammer_cli_katello/repository.rb]: Agreed but I'd want to fix it in the other similar code areas as well. I switched this back to work-in-progress.

Comment (ehelms) [inline lib/hammer_cli_katello/repository.rb]: But you didn't introduce this new code in other areas? If there are similar areas, then I agree make changes in a different PR. This one is a nice small atomic change, that what I am asking about should take 15 minutes to verify and adjust if need be.

**Comment by daviddavis** [comment]
what's the status of this?

**Comment by thomasmckay** [comment]
Status: discouraged. Closing.
